// react-pure-loaders/lib/LineScaleRandom/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const random = 0.5;
const ballBeat = glamor_1.keyframes({
    '0%': {
        transform: 'scale(1.0)',
    },
    '100%': {
        transform: 'scale(1.0)',
    },
    '50%': {
        transform: `scale(${random})`,
    },
});
function lineScale() {
    let i = 1;
    const grid = {};
    for (; i < 5; i += 1) {
        const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
        const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.3);
        grid[` > div:nth-child(${i})`] = {
            animationDelay: `${delay}s`,
            animationDuration: `${duration}s`,
        };
    }
    return grid;
}
exports.default = (color) => glamor_1.css(lineScale(), {
    ' > div': mixins_1.lines(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    ' > div': {
        animationDelay: '0',
        animationIterationCount: 'infinite',
        animationName: ballBeat,
        display: 'inline-block',
    },
});
