// react-pure-loaders/lib/BallScaleRippleMultiple/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const utilities_1 = require("../utilities");
const variables_1 = require("../variables");
const size = '50px';
const ballScaleRippleMultiple = glamor_1.keyframes({
    '0%': {
        opacity: 1,
        transform: 'scale(0.1)',
    },
    '100%': {
        opacity: 0.0,
    },
    '70%': {
        opacity: 0.7,
        transform: 'scale(1)',
    },
});
function ballScaleRipple() {
    const n = 3;
    let i = 0;
    const grid = {};
    for (; i < 4; i += 1) {
        grid[`> div:nth-child(${i})`] = {
            animationDelay: utilities_1.delay('0.2s', n, i - 1),
        };
    }
    return grid;
}
exports.default = (color) => glamor_1.css(ballScaleRipple(), {
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    transform: `translateY(-${parseInt(size, 10) / 2})`,
    ' > div': {
        animation: `${ballScaleRippleMultiple} 1.25s 0s infinite cubic-bezier(.21,.53,.56,.8)`,
        border: `2px solid ${color ? color : variables_1.PRIMARY_COLOR}`,
        borderRadius: '100%',
        height: size,
        left: '-26px',
        position: 'absolute',
        top: '-2px',
        width: size,
    },
});
