// react-pure-loaders/lib/BallClipRotateMultiple/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const variables_1 = require("../variables");
const rotate = glamor_1.keyframes({
    '0%': {
        transform: 'rotate(0deg) scale(1)',
    },
    '100%': {
        transform: 'rotate(360deg) scale(1)',
    },
    '50%': {
        transform: 'rotate(180deg) scale(0.6)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    ' > div': {
        animation: `${rotate} 1s 0s ease-in-out infinite`,
        border: `2px solid ${color ? color : variables_1.PRIMARY_COLOR}`,
        borderBottomColor: 'transparent',
        borderRadius: '100%',
        borderTopColor: 'transparent',
        height: '35px',
        left: '-20px',
        position: 'absolute',
        top: '-20px',
        width: '35px',
        '&:last-child': {
            animationDirection: 'reverse',
            animationDuration: '0.5s',
            // tslint:disable-next-line: max-line-length
            borderColor: `${color ? color : variables_1.PRIMARY_COLOR} transparent ${color ? color : variables_1.PRIMARY_COLOR} transparent`,
            display: 'inline-block',
            height: '15px',
            left: '-10px',
            top: '-10px',
            width: '15px',
        },
    },
});
