// react-pure-loaders/lib/TriangleSkewSpin/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const variables_1 = require("../variables");
const size = '20px';
const triangleSkewSpin = glamor_1.keyframes({
    '100%': {
        transform: 'perspective(100px) rotateX(0) rotateY(0)',
    },
    '25%': {
        transform: 'perspective(100px) rotateX(180deg) rotateY(0)',
    },
    '50%': {
        transform: 'perspective(100px) rotateX(180deg) rotateY(180deg)',
    },
    '75%': {
        transform: 'perspective(100px) rotateX(0) rotateY(180deg)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.globalAnimation(),
}, {
    ' > div': {
        animation: `${triangleSkewSpin} 3s 0s cubic-bezier(.09,.57,.49,.9) infinite`,
        borderBottom: `${size} solid ${color ? color : variables_1.PRIMARY_COLOR}`,
        borderLeft: `${size} solid transparent`,
        borderRight: `${size} solid transparent`,
        height: '0',
        width: '0',
    },
});
