// react-pure-loaders/lib/CubeTransition/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const variables_1 = require("../variables");
const amount = '50px';
const size = '10px';
const ballBeat = glamor_1.keyframes({
    '100%': {
        transform: 'rotate(-360deg)',
    },
    '25%': {
        transform: `translateX(${amount}) scale(0.5) rotate(-90deg)`,
    },
    '50%': {
        transform: `translate(${amount}, ${amount}) rotate(-180deg)`,
    },
    '75%': {
        transform: `translateY(${amount}) scale(0.5) rotate(-270deg)`,
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    transform: `translate(-${parseInt(amount, 10) / 2}, -${parseInt(amount, 10) / 2})`,
    ' > div': {
        animation: `${ballBeat} 1.6s 0s infinite ease-in-out`,
        backgroundColor: color ? color : variables_1.PRIMARY_COLOR,
        height: size,
        left: '-5px',
        position: 'absolute',
        top: '-5px',
        width: size,
        '&:last-child': {
            animationDelay: '-0.8s',
        },
    },
});
