// react-pure-loaders/lib/BallZigZag/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const amount = '30px';
const ballZig = glamor_1.keyframes({
    '100%': {
        transform: 'translate(0, 0)',
    },
    '33%': {
        transform: `translate(-${parseInt(amount, 10) / 2}px, -${amount})`,
    },
    '66%': {
        transform: `translate(${parseInt(amount, 10) / 2}px, -${amount})`,
    },
});
const ballZag = glamor_1.keyframes({
    '100%': {
        transform: 'translate(0, 0)',
    },
    '33%': {
        transform: `translate(${parseInt(amount, 10) / 2}px, ${amount})`,
    },
    '66%': {
        transform: `translate(-${parseInt(amount, 10) / 2}px, ${amount})`,
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.balls(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    transform: `translate(-${parseInt(amount, 10) / 2}px, -${parseInt(amount, 10) / 2}px)`,
    ' > div': {
        left: '-7px',
        marginLeft: (parseInt(amount, 10) / 2),
        position: 'absolute',
        top: '4px',
        '&:first-child': {
            animation: `${ballZig} 0.7s 0s infinite linear`,
        },
        '&:last-child': {
            animation: `${ballZag} 0.7s 0s infinite linear`,
        },
    },
});
