// react-pure-loaders/lib/BallScaleMultiple/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const utilities_1 = require("../utilities");
const size = '60px';
const ballScaleMultiple = glamor_1.keyframes({
    '0%': {
        opacity: '0',
        transform: 'scale(0.0)',
    },
    '100%': {
        opacity: '0',
        transform: 'scale(1.0)',
    },
    '5%': {
        opacity: '1',
    },
});
function ballScale() {
    const n = 3;
    let i = 2;
    const grid = {};
    for (; i < 4; i += 1) {
        grid[`> div:nth-child(${i})`] = {
            animationDelay: utilities_1.delay('0.2s', n, i),
        };
    }
    return grid;
}
exports.ballScale = ballScale;
exports.default = (color) => glamor_1.css(ballScale(), {
    ' > div': mixins_1.balls(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    transform: `translateY(-${size} / 2)`,
    ' > div': {
        animation: `${ballScaleMultiple} 1s 0s linear infinite`,
        height: size,
        left: '-30px',
        margin: '0',
        opacity: '0',
        position: 'absolute',
        top: '0px',
        width: size,
    },
});
