// react-pure-loaders/lib/LineSpinFadeLoader/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const utilities_1 = require("../utilities");
const radius = '20px';
const lineSpinFade = glamor_1.keyframes({
    '100%': {
        opacity: 1,
    },
    '50%': {
        opacity: 0.3,
    },
});
function lineSpinFadeLoader() {
    const n = 8;
    let i = 1;
    const grid = {};
    for (; i < 9; i += 1) {
        const quarter = (parseInt(radius, 10) / 2) + (parseInt(radius, 10) / 5.5);
        if (i === 1) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: 0,
                top: radius,
            };
        }
        else if (i === 2) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: quarter,
                top: quarter,
                transform: 'rotate(-45deg)',
            };
        }
        else if (i === 3) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: radius,
                top: 0,
                transform: 'rotate(90deg)',
            };
        }
        else if (i === 4) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: quarter,
                top: `-${quarter}`,
                transform: 'rotate(45deg)',
            };
        }
        else if (i === 5) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: '0',
                top: `-${radius}`,
            };
        }
        else if (i === 6) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: `-${quarter}`,
                top: `-${quarter}`,
                transform: 'rotate(-45deg)',
            };
        }
        else if (i === 7) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: `-${radius}`,
                top: '0',
                transform: 'rotate(90deg)',
            };
        }
        else if (i === 8) {
            grid[`> div:nth-child(${i})`] = {
                animation: `${lineSpinFade} 1.2s ${utilities_1.delay('0.12s', n, i)} infinite ease-in-out`,
                left: `-${quarter}`,
                top: quarter,
                transform: 'rotate(45deg)',
            };
        }
    }
    return grid;
}
exports.default = (color) => glamor_1.css(lineSpinFadeLoader(), {
    ' > div': mixins_1.lines(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    left: '-4px',
    position: 'relative',
    top: '-10px',
    ' > div': {
        height: '15px',
        position: 'absolute',
        width: '5px',
    },
});
