// react-pure-loaders/lib/BallRotate/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const rotate = glamor_1.keyframes({
    '0%': {
        transform: 'rotate(0deg) scale(1)',
    },
    '100%': {
        transform: 'rotate(360deg) scale(1)',
    },
    '50%': {
        transform: 'rotate(180deg) scale(.6)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.balls(color),
}, {
    ' > div': {
        '&:before, &:after': mixins_1.balls(color),
    },
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    ' > div': {
        position: 'relative',
        '&:first-child': {
            animation: `${rotate} 1s 0s cubic-bezier(.7,-.13,.22,.86) infinite`,
        },
        '&:before, &:after': {
            content: '\'\'',
            opacity: '0.8',
            position: 'absolute',
        },
        '&:before': {
            left: '-28px',
            top: '0px',
        },
        '&:after': {
            left: '25px',
            top: '0px',
        },
    },
});
