// react-pure-loaders/lib/BallPulseRise/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const riseAmount = '30px';
const ballPulseRiseEven = glamor_1.keyframes({
    '0%': {
        transform: 'scale(1.1)',
    },
    '100%': {
        transform: 'scale(1.0)',
    },
    '25%': {
        transform: `translateY(-${riseAmount})`,
    },
    '50%': {
        transform: 'scale(0.4)',
    },
    '75%': {
        transform: `translateY(${riseAmount})`,
    },
    '99%': {
        transform: 'translateY(0)',
    },
});
const ballPulseRiseOdd = glamor_1.keyframes({
    '0%': {
        transform: 'scale(0.4)',
    },
    '100%': {
        transform: 'scale(0.75)',
    },
    '25%': {
        transform: `translateY(${riseAmount})`,
    },
    '50%': {
        transform: 'scale(1.1)',
    },
    '75%': {
        transform: `translateY(-${riseAmount})`,
    },
    '99%': {
        transform: 'translateY(0)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.balls(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    ' > div': {
        animationDelay: 0,
        animationDuration: '1s',
        animationIterationCount: 'infinite',
        animationTimingFunction: 'cubic-bezier(.15,.46,.9,.6)',
        display: 'inline-block',
        '&:nth-child(2n)': {
            animationName: ballPulseRiseEven,
        },
        '&:nth-child(2n-1)': {
            animationName: ballPulseRiseOdd,
        },
    },
});
