// react-pure-loaders/lib/SemiCircleSpin/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const variables_1 = require("../variables");
const size = '35px';
const pos = 30;
const spinRotate = glamor_1.keyframes({
    '0%': {
        transform: 'rotate(0deg)',
    },
    '100%': {
        transform: 'rotate(360deg)',
    },
    '50%': {
        transform: 'rotate(180deg)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.globalAnimation(),
}, {
    height: size,
    overflow: 'hidden',
    position: 'relative',
    width: size,
    ' > div': {
        animation: `${spinRotate} 0.6s 0s infinite linear`,
        // tslint:disable-next-line: max-line-length
        backgroundImage: `linear-gradient(transparent 0%, transparent ${(100 - pos)}%, ${color ? color : variables_1.PRIMARY_COLOR} ${pos}%, ${color ? color : variables_1.PRIMARY_COLOR} 100%)`,
        borderRadius: '100%',
        borderWidth: '0px',
        height: '100%',
        position: 'absolute',
        width: '100%',
    },
});
