// react-pure-loaders/lib/BallClipRotatePulse/style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const mixins_1 = require("../mixins");
const variables_1 = require("../variables");
const rotate = glamor_1.keyframes({
    '0%': {
        transform: 'rotate(0deg) scale(1)',
    },
    '100%': {
        transform: 'rotate(360deg) scale(1)',
    },
    '50%': {
        transform: 'rotate(180deg) scale(0.6)',
    },
});
const scale = glamor_1.keyframes({
    '100%': {
        transform: 'scale(1)',
    },
    '30%': {
        transform: 'scale(0.3)',
    },
});
exports.default = (color) => glamor_1.css({
    ' > div': mixins_1.balls(color),
}, {
    ' > div': mixins_1.globalAnimation(),
}, {
    position: 'relative',
    transform: 'translateY(-15px)',
    ' > div': {
        borderRadius: '100%',
        left: '0px',
        position: 'absolute',
        top: '0px',
        '&:first-child': {
            animation: `${scale} 1s 0s cubic-bezier(.09,.57,.49,.9) infinite`,
            background: color ? color : variables_1.PRIMARY_COLOR,
            height: '16px',
            left: '-7px',
            top: '7px',
            width: '16px',
        },
        '&:last-child': {
            animation: `${rotate} 1s 0s cubic-bezier(.09,.57,.49,.9) infinite`,
            animationDuration: '1s',
            background: 'transparent',
            border: '2px solid',
            // tslint:disable-next-line: max-line-length
            borderColor: `${color ? color : variables_1.PRIMARY_COLOR} transparent ${color ? color : variables_1.PRIMARY_COLOR} transparent`,
            height: '30px',
            left: '-16px',
            position: 'absolute',
            top: '-2px',
            width: '30px',
        },
    },
});
