// react-pure-loaders/lib/mixins/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const glamor_1 = require("glamor");
const variables_1 = require("../variables");
function globalBg(color) {
    return {
        backgroundColor: color ? color : variables_1.PRIMARY_COLOR,
    };
}
exports.globalBg = globalBg;
function globalAnimation() {
    return {
        animationFillMode: 'both',
    };
}
exports.globalAnimation = globalAnimation;
function balls(color) {
    return glamor_1.css({
        borderRadius: '100%',
        height: variables_1.BALL_SIZE,
        margin: variables_1.MARGIN,
        width: variables_1.BALL_SIZE,
    }, globalBg(color));
}
exports.balls = balls;
function ballGrid() {
    let i = 1;
    const grid = {};
    for (; i < 10; i += 1) {
        const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
        const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.6);
        grid[` > div:nth-child(${i})`] = {
            animationDelay: `${delay}s`,
            animationDuration: `${duration}s`,
        };
    }
    return grid;
}
exports.ballGrid = ballGrid;
function lines(color) {
    return glamor_1.css({
        borderRadius: '2px',
        height: variables_1.LINE_HEIGHT,
        margin: variables_1.MARGIN,
        width: variables_1.LINE_WIDTH,
    }, globalBg(color));
}
exports.lines = lines;
